src/CameraControls.ts

```diff
@@ -616,6 +587,51 @@ export class CameraControls extends EventDispatcher {
 
 	}
 
+	get camera() {
+		return this._camera;
+	}
+
+	set camera(camera: _THREE.PerspectiveCamera | _THREE.OrthographicCamera) {
+		this._camera = camera;
+
+		// camera up
+		this._yAxisUpSpace.setFromUnitVectors( this._camera.up, _AXIS_Y );
+		this._yAxisUpSpaceInverse.copy(quatInvertCompat( this._yAxisUpSpace.clone() ));
+
+		// rotation
+		this._spherical.setFromVector3( _v3A.copy( this._camera.position ).applyQuaternion( this._yAxisUpSpace ) );
+		this._sphericalEnd.copy(this._spherical);
+
+		// Zoom
+		this._zoom = this._camera.zoom;
+		this._zoomEnd = this._zoom;
+
+		// collisionTest
+		this._updateNearPlaneCorners();
+
+		// Mouse controls config
+		this.mouseButtons = {
+			left: ACTION.ROTATE,
+			middle: ACTION.DOLLY,
+			right: ACTION.TRUCK,
+			wheel:
+				isPerspectiveCamera( this._camera )  ? ACTION.DOLLY :
+				isOrthographicCamera( this._camera ) ? ACTION.ZOOM :
+				ACTION.NONE,
+			// We can also add shiftLeft, altLeft and etc if someone wants...
+		};
+
+		// Touch controls config
+		this.touches = {
+			one: ACTION.TOUCH_ROTATE,
+			two:
+				isPerspectiveCamera( this._camera )  ? ACTION.TOUCH_DOLLY_TRUCK :
+				isOrthographicCamera( this._camera ) ? ACTION.TOUCH_ZOOM_TRUCK :
+				ACTION.NONE,
+			three: ACTION.TOUCH_TRUCK,
+		};
+	}
```

Comment:
Thanks for organizing the PR.
However, I realized that `set camera` can be just a setter of the `_camera` without any side effects.

```suggestion
	set camera( camera: _THREE.PerspectiveCamera | _THREE.OrthographicCamera ) {
		this._camera = camera;
		this._needsUpdate = true;
	}
```

And other setting changes can be done in the user space.

e.g.
```
cameraControls.camera = anotherCamera;
// touches.two have to be changed, but I don't want to change for touches.one;
cameraControls.touches.two = whatever;
```

Reply:
For the `touches` and `mouseButtons`, I agree with you, they're already exposed to the user.

> I realized that set camera can be just a setter of the _camera without any side effects.

Are you sure about that, I mean what if the new camera has a different `position`, `up` or `zoom`, especially with `OrthographicCamera` where the zoom is important.

If you want to remove those settings anyway, it's fine by me, I'll commit you're updates.

Thanks

Reply:
If users need to change `position`, `zoom`, and other props, I think the user can make another `CameraControls` instance rather than reusing it.

It would be nice if the user could freely set whether the position and zoom should move immediately or smoothly.
However, if the `set camera` changes the position or zoom, it cannot be handled.